api/src/opentrons/protocol_api/core/engine/instrument.py

```diff
@@ -721,33 +724,33 @@ def get_pipette_name(self) -> str:
 
         Will match the load name of the actually loaded pipette,
         which may differ from the requested load name.
-        """
-        # TODO (tz, 11-23-22): revert this change when merging
-        # https://opentrons.atlassian.net/browse/RLIQ-251
-        pipette = self._engine_client.state.pipettes.get(self._pipette_id)
-        return (
-            pipette.pipetteName.value
-            if isinstance(pipette.pipetteName, PipetteNameType)
-            else pipette.pipetteName
-        )
 
-    def get_load_name(self) -> str:
-        """Get the pipette's requested API load name.
+        From API v2.15 to v2.22, this property returned an internal, engine-specific,
+         name for Flex pipettes (eg, "p50_multi_flex" instead of "flex_8channel_50").
 
-        This is the load name that is specified in the `ProtocolContext.load_instrument()`
-        method. This name might differ from the engine-specific pipette name.
+        From API v2.23 onwards, this behavior is fixed so that this property returns
+        the API-specific names of Flex pipettes.
         """
+        # TODO (tz, 11-23-22): revert this change when merging
+        # https://opentrons.atlassian.net/browse/RLIQ-251
         pipette = self._engine_client.state.pipettes.get(self._pipette_id)
-        load_name = next(
-            (
-                pip_api_name
-                for pip_api_name, pip_name in PIPETTE_API_NAMES_MAP.items()
-                if pip_name == pipette.pipetteName
-            ),
-            None,
-        )
-        assert load_name, "Load name not found."
-        return load_name
+        if self._protocol_core.api_version < _FLEX_PIPETTE_NAMES_FIXED_IN:
+            return (
+                pipette.pipetteName.value
+                if isinstance(pipette.pipetteName, PipetteNameType)
+                else pipette.pipetteName
+            )
```

Comment:
As far as Pyright and I can tell, this `if isinstance(...)` call is always true because `pipette.pipetteName` is always a `pipetteNameType`, right? Can this just be:

```suggestion
            return pipette.pipetteName.value
```

Reply:
Ya I was wondering about this but didn't want to change anything in case I was overlooking something. This is from the existing function. Let's see if we break something.